### DISCUSSION TITLE
Multiple, unrelated CTEs found

### DISCUSSION THREAD

**Discussion by pdc1** [sqlalchemy/sqlalchemy#10628] [Usage Questions] [answered]
Hi all,

See my [previous question](https://github.com/sqlalchemy/sqlalchemy/discussions/10623) for background, but I have a query with CTE and subquery and I wanted to switch it to an ORM result.

From my previous question, the original (working) "core" query looks like this:
```python
sq = (
    select(BookDB.author, BookDB.date_read, BookDB.title)
    .where(BookDB.author == "<author>")
    .cte()
)
sq2 = (
    select(sq.c.author, func.max(sq.c.date_read))
    .subquery()
)
q = (
    select(sq.c.author, sq.c.date_read, sq.c.title, sq2)
    .select_from(sq)
    .join(sq2, sq.c.author == sq2.c.author)
    .where(sq.c.title == "<title>")
)
with Session(engine) as session:
    row = session.execute(q).one_or_none()
    if row:
        print(">", row)
```

I wanted to update this to return the ORM BookDB object. My understanding is I need to alias the BookDB object to the CTE, but now  get an error: `CompileError: Multiple, unrelated CTEs found with the same name: 'anon_2'`
```python
cte = (
    select(BookDB)
    .where(BookDB.author == "<author>")
    .cte()
)
book_cte = aliased(BookDB, cte)
sq2 = (
    select(book_cte.author, func.max(book_cte.date_read))
    .subquery()
)
q = (
    select(BookDB, sq2)
    .select_from(book_cte)
    .join(sq2, book_cte.author == sq2.c.author)
    .where(book_cte.title == "<title>")
)
with Session(engine) as session:
    row = session.execute(q).one_or_none()
    if row:
        print(">", row)
```

I'm guessing part of the problem is the sub-query (sq2) is returning a function result and not an ORM object.

This may be another indication that I am trying too hard with the CTE+sub-query, and should just have two queries, one for ORM BookDB results, and one core query for the max date function. But I thought I'd throw the question out there in case I'm missing something.

Thanks!

**Comment by CaselIT** [comment, marked as answer]
Hi,

Try changing this `select(BookDB, sq2)...` to this `select(book_cte, sq2)...` so that the alias is used also in the select. It should still load `BookDB` objects

**Comment by pdc1** [comment]
So initially I still got the error, but then I thought to remove the `.select_from(book_cte)` now that the select is on book_cte, and that seems to work!
